src/js/lib/monster.ui.js

```diff
@@ -3468,6 +3468,223 @@ define(function(require) {
 		}
 	};
 
+	/**
+	 * @param  {Object} args
+	 * @param  {Object[]} [args.choices]
+	 * @param  {Object} [args.existing]
+	 * @param  {Boolean} [args.isEditable=false]
+	 * @param  {Object} [args.i18n]
+	 * @param  {Function} [args.normalizer]
+	 * @param  {Function} [args.onSelect]
+	 */
+	function keyValueSelector(pArgs) {
+		var self = monster.apps.core,
+			args = _.merge({
+				choices: [],
+				existing: {},
+				isEditable: false,
+				i18n: {
+					title: self.i18n.active().keyValueSelector.title
+				},
+				normalizer: _.snakeCase
+			}, pArgs),
+			choices = args.choices,
+			existing = args.existing,
+			isEditable = _.isEmpty(choices) ? true : args.isEditable,
+			i18n = args.i18n,
+			normalizer = args.normalizer,
+			onSelect = args.onSelect,
+			$template = $(self.getTemplate({
+				name: 'monster-keyValueSelector',
+				data: {
+					i18nCustom: _.get(args, 'i18n', {}),
+					choices: choices,
+					isEditable: isEditable
+				}
+			})),
+			$container = monster.ui.dialog($template, _.merge({
+				autoScroll: false
+			}, _.pick(i18n, [
+				'title'
+			]))),
+			$form = $container.find('#key_value_selector_form'),
+			itemsPerCategory = _.mergeWith(
+				_
+					.chain(existing)
+					.keyBy(function(v, category) {
```

Comment:
`lodash#keyBy()` iteratee function only receives one parameter, so `category` is always `undefined`. I think this function should be replaced by `lodash#mapKeys()`.

Reply:
Good catch! 💪